tf/main.tf

```diff
@@ -0,0 +1,89 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+provider "ec" {}
+
+data "ec_stack" "latest" {
+  version_regex = "latest"
+  region        = var.ec_region
+}
+
+resource "ec_deployment" "ec_aws_lambda_minimal" {
+  name                   = "aws-lambda-smoke-testing-deployment"
+  region                 = var.ec_region
+  version                = data.ec_stack.latest.version
+  deployment_template_id = var.ec_deployment_template
+
+  elasticsearch {
+    autoscale = "false"
+
+    topology {
+      id         = "hot_content"
+      size       = "1g"
+      zone_count = 1
+    }
+  }
+}
+
+resource "ec_deployment" "basic_datasource" {
+  name                   = "aws-lambda-smoke-testing-deployment-data"
+  region                 = var.ec_region
+  version                = data.ec_stack.latest.version
+  deployment_template_id = var.ec_deployment_template
+
+  elasticsearch {
+    topology {
+      id         = "hot_content"
+      size       = "1g"
+      zone_count = 1
+    }
+  }
+
+  kibana {}
+
+  apm {}
```

Comment:
This is legacy, and will eventually be removed, so we should be using "integrations_server" here instead. The problem then becomes how to fetch the generated secret token - hence ec_deployment.

Reply:
Solved by switching to APM server custom module